src/pages/dataElements/form/hooks.ts

```diff
@@ -0,0 +1,45 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { useMemo } from 'react'
+import { Attribute } from '../../../types/generated'
+
+const CUSTOM_ATTRIBUTES_QUERY = {
+    attributes: {
+        resource: 'attributes',
+        params: {
+            fields: [
+                'id',
+                'mandatory',
+                'displayFormName',
+                'valueType',
+                'optionSet[*, options[id,displayName,name,code]]',
+            ],
+            paging: false,
+            filter: 'dataElementAttribute:eq:true',
+        },
+    },
+}
+
+// @TODO(useCustomAttributesQuery): create response type
+interface QueryResponse {
+    attributes: {
+        attributes: Attribute[]
+    }
+}
+
+export function useCustomAttributesQuery() {
```

Comment:
I think it should be somewhat trivial to create a generic version of this. Don't have to do that now, but worth a note when looking at other sections.